faebryk/library/library/footprints.py

```diff
@@ -26,6 +26,22 @@ def get_kicad_footprint() -> str:
 
         self.add_trait(_has_kicad_footprint())
 
+class QFN(Footprint):
+    def __init__(self, pin_cnt: int, size_x_mm: int, size_y_mm: int) -> None:
+        super().__init__()
+
+        class _has_kicad_footprint(has_kicad_footprint):
+            @staticmethod
+            def get_kicad_footprint() -> str:
+                return \
+                    "Package_QFN:QFN-{leads}_X{spacing_x:.2f}mm_Y{spacing_y:.2f}mm".format(
```

Comment:
According to https://kicad.github.io/footprints/Package_DFN_QFN,
these are the options we need to support (example):
QFN-16-1EP_4x4mm_P0.5mm_EP2.45x2.45mm
QFN-16-1EP_4x4mm_P0.5mm_EP2.45x2.45mm_ThermalVias

- 16 = pin_cnt
- 1EP = exposed thermal pad count
- 4x4 = package X/Y size mm
- P0.5 = lead/pin pitch
- EP2.45x2.45mm = exposed thermal pad size XxYmm
- ThermalVias = yes/no thermal vias in exposed thermal pad